pyam_analysis/core.py

```diff
@@ -65,71 +65,94 @@ class IamDataFrame(object):
     (including validation of values, completeness of variables provided)
     as well as a number of visualization and plotting tools."""
 
-    def __init__(self, data, regions=None, **kwargs):
+    def __init__(self, data, regions=None, variables=None, units=None,
```

Comment:
if I read this correctly, you can actually get rid of all of these ix kwargs. If you simply use **kwargs, you can do read_ix(..., **kwargs) and it will pass straight through.